dbms/src/IO/ReadHelpers.h

```diff
@@ -302,32 +309,40 @@ ReturnType readIntTextImpl(T & x, ReadBuffer & buf)
                 res += *buf.position() - '0';
                 break;
             default:
-                x = negative ? -res : res;
-                return ReturnType(true);
+                goto end;
+//                x = negative ? -res : res;
+//                return ReturnType(true);
         }
         ++buf.position();
     }
 
+end:
     x = negative ? -res : res;
+    if (check_overflow && buf.count() - initial > std::numeric_limits<T>::digits10)
```

Comment:
This check is only for decimal overflow. E.g. `std::numeric_limits<char>::digits10` is 2, but number `111` can be represented in char.

Reply:
I agree, but correct overflow check would require to use `mulOverflow`, `addOverflow` and additional runtime checks on last few digits, roughly:
```
            case '9':
                if constexpr (check_overflow == ReadIntTextCheckOverflow::CHECK_OVERFLOW)
                {
                    // perform relativelly slow overflow check only when number of decimal digits so far is close to the max for given type.
                    if (buf.count() - initial_pos >= std::numeric_limits<T>::max_digits10)
                    {
                        if (common::mulOverflow(res, static_cast<decltype(res)>(10), res)
                            || common::addOverflow(res, static_cast<decltype(res)>(*buf.position() - '0'), res))
                            return ReturnType(false);
                        break;
                    }
                }
                res *= 10;
                res += *buf.position() - '0';
                break;
```

Reply:
I mean, it's good that we have this check, and it's perfect for decimals.
Just maybe rename it to `CHECK_DECIMAL_OVERFLOW`, to specify what we really check.